### PR TITLE
fix: add support for showing approval voting results

### PR DESCRIPTION
**Pull request by wa0x6e**
### Summary

Add support for showing vote result for approval voting type (in the proposal's "vote" tab)

### How to test

1. Go to a proposal with `approval` voting type (e.g. http://localhost:8080/#/s:pistachiodao.eth/proposal/0x08c3bd2960700525770a1d634f8599ba967e55fcc05b6c1649d984d88253769d/votes)
2. It should show all the user's choices
3. It should show the full choices string in the tooltip, in case the string is truncated inside the table

Screenshot 2024-03-15 at 14 39 57

It should keep showing the old format for the existing vote type:

Single choice:
Screenshot 2024-03-15 at 14 40 03

Basic:
Screenshot 2024-03-15 at 14 40 09

The VotesModal should also show the same choice text, depending on voting type, albeit may need some UI tweaking

Screenshot 2024-03-15 at 15 41 17

### CODE REVIEW

**Comment by Sekhmet** [comment]
It's missing ModalVotes updates.

**Comment by wa0x6e** [comment]
> It's missing ModalVotes updates.

In which area can I see the button to open this modal ?

**Comment by wa0x6e** [comment]
> It's missing ModalVotes updates.

Added

File: apps/ui/src/components/Modal/Votes.vue [hunk 1]
```diff
@@ -48,6 +48,14 @@ async function handleEndReached() {
   loadingMore.value = false;
 }
 
+function getChoiceText(proposalType: VoteType, choice: Choice) {
```
Comment (Sekhmet) [inline apps/ui/src/components/Modal/Votes.vue:51]: This isn't `Choice`, it's either `number | number[]`.

Also we are now duplicating it in two files, can we move it to helpers file like there?
https://github.com/snapshot-labs/sx-monorepo/pull/174#issuecomment-1997578149

**Review by Sekhmet — COMMENTED** [review]
Other than that it looks good!

**Review by Sekhmet — APPROVED** [review]
tACK

**Pull request merged**